addons/guide/modifiers/guide_modifier_map_range.gd

```diff
@@ -0,0 +1,40 @@
+@tool
+## Maps an input range to an output range and optionally clamps the output.
+class_name GUIDEModifierMapRange
+extends GUIDEModifier
+
+## Should the output be clamped to the range?
+@export var apply_clamp:bool = false
+
+## The minimum input value
+@export var input_min:float = 0.0
+
+## The maximum input value
+@export var input_max:float = 1.0
+
+## The minimum output value
+@export var output_min:float = 0.0
+
+## The maximum output value
+@export var output_max:float = 1.0
+
```

Comment:
I think an axis selector would also be useful here, so we can restrict which axes of the input vector are affected by this modifier, e.g.:

```gdscript
@export var x:bool = true
@export var y:bool = true
@export var z:bool = true
```

This allows for more flexibility, e.g. the user could chain multiple range mappings if they want to apply different ranges to different axes.